lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -818,6 +818,11 @@ def manage_tags(match, new_tags, purge_tags, ec2):
 
 def build_volume_spec(params):
     volumes = params.get('volumes') or []
+    for volume in volumes:
+        if 'ebs' in volume:
+            for int_value in ['volume_size', 'iops']:
+                if int_value in volume['ebs']:
+                    volume['ebs'][int_value] = int(volume['ebs'][int_value])
```

Comment:
Can suboptions handle this? It seems preferable if this was defined via the argspec rather than casting to an int where used.

Reply:
I have tried adding the argspec validation, but it doesn't seem to work, have a look here https://gist.github.com/Shaps/b0430ce00787cd62a14bcd5b16358fc6 for the change I have made.
I just get the same validation error back from boto>

Reply:
Sorry for bugging, this is now in devel branch, but what would be needed for it to make it into a release version..?

Reply:
@dnmvisser - it will end up in 2.9 by default, but to end up in the next 2.8 release, this would need backporting: https://docs.ansible.com/ansible/latest/community/development_process.html#backport-pull-request-process

You can also just use ec2_instance from source without updating ansible itself: http://willthames.github.io/2017/12/12/using-updated-modules-with-stable-ansible.html

Reply:
@dnmvisser I've raised the backport PR in #59687